janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java

```diff
@@ -373,13 +378,21 @@ private Graph writeMutatedPropertiesBackIntoGraph() {
             }
 
             //TODO: Filter based on VertexProgram
+            HaltedTraverserStrategy haltedTraverserStrategy = HaltedTraverserStrategy.reference();
             Map<Long, Map<String, Object>> mutatedProperties = Maps.transformValues(vertexMemory.getMutableVertexProperties(),
-                new Function<Map<String, Object>, Map<String, Object>>() {
-                    @Nullable
-                    @Override
-                    public Map<String, Object> apply(final Map<String, Object> o) {
-                        return Maps.filterKeys(o, s -> !VertexProgramHelper.isTransientVertexComputeKey(s, vertexProgram.getVertexComputeKeys()));
-                    }
+                (Function<Map<String, Object>, Map<String, Object>>) o -> {
+                    Map<String, Object> nonTransientKeys =
```

Comment:
(nitpick) I tried to understand what this function does but failed. If you can move some of this into its own method and give that a good name that tells what the code does, then that would be helpful to any future reader in my opinion.
However, all the Fulgora code is written this way so this added code fits in nicely here 😅